dpnp/dpnp_iface_arraycreation.py

```diff
@@ -1059,7 +1059,9 @@ def trace(arr, offset=0, axis1=0, axis2=1, dtype=None, out=None):
             pass
         elif axis2 != 1:
             pass
-        elif out != None:
+        elif dtype is not None:
+            pass
+        elif out is not None:
```

Comment:
You discard cases where `out.dtype` matches with `arr.dtype`.

Reply:
I think it is acceptable for this stage of development